src/app/common/messagedisplay.ts

```diff
@@ -36,20 +37,38 @@ export abstract class MessageDisplay {
     return this.rows.length;
   }
 
+  selectedRowIds(): number[] {
+    return Object.keys(this.selectedMessageIds).map((msgIndex) => this.findRowByMessageId(parseInt(msgIndex, 10)));
+  }
+
+  // msgIds where the value is true
   selectedMessageIds(): number[] {
-    return Object.keys(this.selectedRowIds).map((rowIndex) => this.getRowMessageId(parseInt(rowIndex, 10)));
+    return Object.keys(this.msgIdsSelected).filter((msgId) => this.msgIdsSelected[parseInt(msgId, 10)]).map((msgId) => parseInt(msgId, 10));
   }
 
   allSelected(): boolean {
-    return Object.keys(this.selectedRowIds).filter((key) => this.selectedRowIds[key]).length === this.rows.length;
+    return Object.keys(this.msgIdsSelected).filter((key) => this.msgIdsSelected[key]).length === this.rows.length;
+  }
+
+  anySelected(): boolean {
+    return Object.keys(this.msgIdsSelected).reduce((prev, current) =>
```

Comment:
I think this could be simplified to `!!Object.keys(this.msgIdsSelected).find(id => this.msgIdsSelected)` or something similar.

Reply:
Or even `this.selectedMessageIds().length > 0`

Reply:
Could even just be a "find where any is true" .. I just moved this code from app.component ... didnt re-read it ;)

Reply:
Object.values(this.msgIdsSelected).includes(true); ? feels like the shortest one